src/api/identity/group/requests.rs

```diff
@@ -0,0 +1,223 @@
+/// * Method: POST
+/// * Reference: https://developer.hashicorp.com/vault/api-docs/secret/identity/group#create-a-group
+#[derive(Builder, Debug, Default, Endpoint, Serialize)]
+#[endpoint(
+    path = "identity/group",
+    method = "POST",
+    builder = "true",
+    response = "CreateGroupResponse"
+)]
+#[builder(setter(into, strip_option), default)]
+pub struct CreateGroupRequest {
+    /// Name of the group. If set (and ID is not set), updates the corresponding existing group.
+    pub name: Option<String>,
+    /// ID of the group. If set, updates the corresponding existing group.
+    pub id: Option<String>,
+    /// Type of the group, internal or external. Defaults to internal.
+    #[serde(rename = "type")]
+    pub group_type: Option<String>,
+    /// Metadata to be associated with the entity.
+    pub metadata: Option<HashMap<String, String>>,
+    /// Policies to be tied to the group.
+    pub policies: Option<Vec<String>>,
+    /// Group IDs to be assigned as group members.
+    pub member_group_ids: Option<Vec<String>>,
+    /// Entity IDs to be assigned as group members.
+    pub member_entity_ids: Option<Vec<String>>,
+}
+
+/// ## Read group by ID
+///
+/// This endpoint queries the group by its identifier.
+///
+/// * Path: identity/group/id/{self.id}
+/// * Method: GET
+/// * Response: [ReadGroupByIdResponse]
+/// * Reference: https://developer.hashicorp.com/vault/api-docs/secret/identity/group#read-group-by-id
+#[derive(Builder, Debug, Endpoint)]
+#[endpoint(
+    path = "identity/group/id/{self.id}",
+    method = "GET",
+    builder = "true",
+    response = "ReadGroupByIdResponse"
+)]
+#[builder(setter(into))]
+pub struct ReadGroupByIdRequest {
+    /// Identifier of the group.
+    #[endpoint(skip)]
+    pub id: String,
+}
+
+/// ## Update group by ID
+///
+/// This endpoint is used to update a existing group.
+///
+/// * Path: identity/group/id/{self.id}
+/// * Method: POST
+/// * Reference: https://developer.hashicorp.com/vault/api-docs/secret/identity/group#update-group-by-id
+#[derive(Builder, Debug, Default, Endpoint, Serialize)]
+#[endpoint(
+    path = "identity/group/id/{self.id}",
+    method = "POST",
+    builder = "true"
+)]
+#[builder(setter(into, strip_option), default)]
+pub struct UpdateGroupByIdRequest {
+    /// Identifier of the group.
+    #[endpoint(skip)]
+    pub id: String,
+    /// Name of the group.
+    pub name: Option<String>,
+    /// Type of the group, internal or external. Defaults to internal.
+    #[serde(rename = "type")]
+    pub group_type: Option<String>,
+    /// Metadata to be associated with the group.
+    pub metadata: Option<HashMap<String, String>>,
+    /// Policies to be tied to the group.
+    pub policies: Option<Vec<String>>,
+    /// Group IDs to be assigned as group members.
+    pub member_group_ids: Option<Vec<String>>,
+    /// Entity IDs to be assigned as group members.
+    pub member_entity_ids: Option<Vec<String>>,
+}
+
+/// ## Delete group by ID
```

Comment:
Also the TryFrom like in the entity request.
```suggestion
}

impl TryFrom<CreateGroupRequest> for UpdateGroupByIdRequest {
    type Error = &'static str;

    fn try_from(value: CreateGroupRequest) -> Result<Self, Self::Error> {
        match value.id {
            Some(id) => Ok(Self { id, name: value.name, group_type: value.group_type, metadata: value.metadata, policies: value.policies, member_group_ids: value.member_group_ids, member_entity_ids: value.member_entity_ids}),
            None => Err("The id attribute has to be some"),
        }
    }
}

/// ## Delete group by ID
```